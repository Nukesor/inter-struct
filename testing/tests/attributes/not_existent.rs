use inter_struct::prelude::*;

/// Ensure that error messages for each missing attribute are shown
#[derive(StructInto, StructMerge, StructMergeRef, StructIntoDefault)]
pub struct FromStruct {
    pub normal: String,
}

pub struct IntoStruct {
    pub normal: String,
    pub optional: Option<String>,
}

fn main() {}
