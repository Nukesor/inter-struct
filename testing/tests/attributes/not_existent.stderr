error: StructInto requires the 'struct_into' attribute.
 --> tests/attributes/not_existent.rs:5:12
  |
5 | pub struct FromStruct {
  |            ^^^^^^^^^^

error: StructMerge requires the 'struct_merge' attribute.
 --> tests/attributes/not_existent.rs:5:12
  |
5 | pub struct FromStruct {
  |            ^^^^^^^^^^

error: StructMergeRef requires the 'struct_merge_ref' attribute.
 --> tests/attributes/not_existent.rs:5:12
  |
5 | pub struct FromStruct {
  |            ^^^^^^^^^^

error: StructInto requires the 'struct_into_default' attribute.
 --> tests/attributes/not_existent.rs:5:12
  |
5 | pub struct FromStruct {
  |            ^^^^^^^^^^
